### DISCUSSION TITLE
Date picker: Select multiple, non-contiguous dates

### DISCUSSION THREAD

**Discussion by brielov** [adobe/react-spectrum#4388] [Q&A] [answered]
I'm trying to migrate one of my company's legacy date picker components to use react-aria (mostly for accessibility issues). This legacy implementation accepts an array of non-contiguous dates, meaning you can select arbitrary dates that are not necessarily a range. Can this be done with react-aria?

**Comment by reidbarber** [comment]
Yes, if you're interested in using the `useDatePicker` hook to build it: https://react-spectrum.adobe.com/react-aria/useDateRangePicker.html#non-contiguous-ranges

Or use the new headless `DatePicker` component: https://react-spectrum.adobe.com/react-aria/DateRangePicker.html#non-contiguous-ranges

**Reply by brielov** [reply]
@reidbarber The thing is that this still gives me a range. My current component allows you to select multiple, individual, non-contiguous dates. Like you can click next monday and tuesday, but you can also click next month's thursday and friday. You end up with an array of dates that are not in a range nor share any context between them. Is this doable? if is not, is there any way I can still use the accessibility perks of react-aria while maintaining my own selected dates logic?

**Reply by reidbarber** [reply, marked as answer]
Ahh ok, I misunderstood. I believe I answered this here then: https://github.com/adobe/react-spectrum/discussions/3657#discussioncomment-3908594
